generators/react-native/resources/expo/package.json

```
@@ -7,35 +7,36 @@
     "update-dependencies": "npm install && npx expo install --fix && rm -rf package-lock.json node_modules"
   },
   "dependencies": {
-    "@expo/vector-icons": "^13.0.0",
-    "@react-native-async-storage/async-storage": "1.18.2",
-    "@react-native-community/datetimepicker": "7.2.0",
-    "@react-native-masked-view/masked-view": "0.2.9",
-    "@react-native-picker/picker": "2.4.10",
+    "@expo/metro-runtime": "~3.2.1",
+    "@expo/vector-icons": "^14.0.0",
+    "@react-native-async-storage/async-storage": "1.23.1",
+    "@react-native-community/datetimepicker": "8.0.1",
+    "@react-native-masked-view/masked-view": "0.3.1",
+    "@react-native-picker/picker": "2.7.5",
     "@react-navigation/devtools": "6.0.26",
-    "expo": "49.0.21",
-    "expo-auth-session": "~5.0.2",
-    "expo-constants": "~14.4.2",
-    "expo-image-picker": "~14.3.2",
-    "expo-linking": "~5.0.2",
-    "expo-random": "~13.2.0",
-    "expo-splash-screen": "~0.20.5",
-    "expo-status-bar": "~1.6.0",
-    "expo-web-browser": "~12.3.2",
+    "expo": "51.0.11",
+    "expo-auth-session": "~5.5.2",
+    "expo-constants": "~16.0.2",
+    "expo-image-picker": "~15.0.5",
+    "expo-linking": "~6.3.1",
+    "expo-random": "~14.0.1",
+    "expo-splash-screen": "~0.27.5",
+    "expo-status-bar": "~1.12.1",
+    "expo-web-browser": "~13.0.3",
     "react": "18.2.0",
     "react-dom": "18.2.0",
-    "react-native": "0.72.10",
-    "react-native-gesture-handler": "~2.12.0",
-    "react-native-reanimated": "~3.3.0",
-    "react-native-safe-area-context": "4.6.3",
-    "react-native-screens": "~3.22.0",
+    "react-native": "0.74.2",
+    "react-native-gesture-handler": "~2.16.1",
+    "react-native-reanimated": "~3.10.1",
+    "react-native-safe-area-context": "4.10.1",
+    "react-native-screens": "3.31.1",
     "react-native-web": "~0.19.12"
   },
   "devDependencies": {
     "@babel/core": "^7.24.7",
     "@react-native-community/cli-server-api": "13.6.9",
     "@types/react": "~18.2.65",
-    "jest-expo": "^49.0.0",
-    "typescript": "^5.4.5"
+    "jest-expo": "~51.0.1",
+    "typescript": "~5.3.3"
```

```suggestion
    "typescript": "^5.4.5"
```
Don't downgrade TypeScript